Rename Storage class name in TheWarehouse to avoid conflict compiling with Torch

### Motivation

When I tried to compile my application with Torch, I got compiling errors like
```
/Users/xxx/mambaforge3/envs/moose//lib/python3.10/site-packages/torch//include/ATen/core/DeprecatedTypeProperties.h:127:3: error: reference to 'Storage' is ambiguous
  Storage unsafeStorageFromTH(void * th_pointer, bool retain) const;
  ^
/Users/xxx/projects/check/myapp/moose/framework/build/header_symlinks/TheWarehouse.h:24:7: note: candidate found by name lookup is 'Storage'
class Storage;
      ^
/Users/xxx/mambaforge3/envs/moose//lib/python3.10/site-packages/torch//include/c10/core/Storage.h:25:16: note: candidate found by name lookup is 'c10::Storage'
struct C10_API Storage {
               ^
```
I attempted to rename `Storage` with something like `WHStorage`, this compiling error is gone.

### Design

This appears a part of a known broader issue noted in https://mooseframework.inl.gov/getting_started/installation/install_libtorch.html. The better fix is possibly using namespace. But for now renaming this `Storage` class, which is only used by `TheWarehouse` can solve my immediate issue. And the fix is minor.

### Impact

Make codes using Torch easier to be compiled.

--- a/framework/include/base/TheWarehouse.h
+++ b/framework/include/base/TheWarehouse.h
@@ -21,7 +21,7 @@
 #include "MooseUtils.h"
 
 class MooseObject;
-class Storage;
+class WarehouseStorage;
 class TheWarehouse;
 
 /// Attribute is an abstract class that can be implemented in order to track custom metadata about
@@ -515,7 +515,7 @@ private:
 
   void readAttribs(const MooseObject * obj, std::vector<std::unique_ptr<Attribute>> & attribs);
 
-  std::unique_ptr<Storage> _store;
+  std::unique_ptr<WarehouseStorage> _store;
   std::vector<std::shared_ptr<MooseObject>> _objects;
   std::unordered_map<MooseObject *, std::size_t> _obj_ids;
 

--- a/framework/src/base/TheWarehouse.C
+++ b/framework/src/base/TheWarehouse.C
@@ -18,10 +18,10 @@
 
 #include <memory>
 
-class Storage
+class WarehouseStorage
 {
 public:
-  virtual ~Storage() = default;
+  virtual ~WarehouseStorage() = default;
   virtual void add(std::size_t obj_id, std::vector<std::unique_ptr<Attribute>> attribs) = 0;
   virtual std::vector<std::size_t> query(const std::vector<std::unique_ptr<Attribute>> & conds) = 0;
   virtual void set(std::size_t obj_id, std::vector<std::unique_ptr<Attribute>> attribs) = 0;
@@ -53,7 +53,7 @@ AttribSorted::isEqual(const Attribute & other) const
   return isMatch(other);
 }
 
-class VecStore : public Storage
+class VecStore : public WarehouseStorage
 {
 public:
   virtual void add(std::size_t obj_id, std::vector<std::unique_ptr<Attribute>> attribs) override

